src/agentscope/rag/_store/_qdrant_store.py

```diff
@@ -0,0 +1,163 @@
+    ) -> None:
+        """Initialize the local Qdrant vector store.
+
+        Args:
+            location (`Literal[":memory:"] | str`):
+                The location of the Qdrant instance. Use ":memory:" for
+                in-memory Qdrant instance, or url for remote Qdrant instance,
+                e.g. "http://localhost:6333" or a path to a directory.
+            collection_name (`str`):
+                The name of the collection to store the embeddings.
+            dimensions (`int`):
+                The dimension of the embeddings.
+            distance (`Literal["Cosine", "Euclid", "Dot", "Manhattan"]`, \
+            default to "Cosine"):
+                The distance metric to use for the collection. Can be one of
+                "Cosine", "Euclid", "Dot", or "Manhattan". Defaults to
+                "Cosine".
+            client_kwargs (`dict[str, Any] | None`, optional):
+                Other keyword arguments for the Qdrant client.
+            collection_kwargs (`dict[str, Any] | None`, optional):
+                Other keyword arguments for creating the collection.
+        """
+
+        try:
+            from qdrant_client import AsyncQdrantClient
+        except ImportError as e:
+            raise ImportError(
+                "Qdrant client is not installed. Please install it with "
+                "`pip install qdrant-client`.",
+            ) from e
+
+        client_kwargs = client_kwargs or {}
+        self._client = AsyncQdrantClient(location=location, **client_kwargs)
+
+        self.collection_name = collection_name
+        self.dimensions = dimensions
+        self.distance = distance
+        self.collection_kwargs = collection_kwargs or {}
+
+    async def _validate_collection(self) -> None:
+        """Validate the collection exists, if not, create it."""
+        if not await self._client.collection_exists(self.collection_name):
+            from qdrant_client import models
+
+            collections_kwargs = {
+                "collection_name": self.collection_name,
+                "vectors_config": models.VectorParams(
+                    size=self.dimensions,
+                    distance=getattr(models.Distance, self.distance.upper()),
+                ),
+                **self.collection_kwargs,
+            }
+            await self._client.create_collection(**collections_kwargs)
+
+    async def add(self, documents: list[Document], **kwargs: Any) -> None:
+        """Add embeddings to the Qdrant vector store.
+
+        Args:
+            documents (`list[Document]`):
+                A list of embedding records to be recorded in the Qdrant store.
+        """
+        await self._validate_collection()
+
+        from qdrant_client.models import PointStruct
+
+        await self._client.upsert(
+            collection_name=self.collection_name,
+            points=[
+                PointStruct(
+                    id=uuid.uuid4().hex,
```

Comment:
It seems that each addition generates a new ID, which means adding the same document multiple times creates multiple entries rather than overwriting. Is this the intended behavior?

Reply:
Update in the latest commit